main.go

```diff
@@ -111,15 +113,48 @@ func run(cfg langserver.Config) error {
 		}
 		defer lis.Close()
 
-		log.Println("langserver-go: listening on", *addr)
+		log.Println("langserver-go: listening for TCP connections on", *addr)
+
+		connectionCount := 0
+
 		for {
 			conn, err := lis.Accept()
 			if err != nil {
 				return err
 			}
-			jsonrpc2.NewConn(context.Background(), jsonrpc2.NewBufferedStream(conn, jsonrpc2.VSCodeObjectCodec{}), handler, connOpt...)
+			connectionCount = connectionCount + 1
+			connectionID := connectionCount
+			log.Printf("langserver-go: received incoming connection #%d\n", connectionID)
+			jsonrpc2Connection := jsonrpc2.NewConn(context.Background(), jsonrpc2.NewBufferedStream(conn, jsonrpc2.VSCodeObjectCodec{}), handler, connOpt...)
+			go func() {
+				<-jsonrpc2Connection.DisconnectNotify()
+				log.Printf("langserver-go: disconnected connection #%d\n", connectionID)
+			}()
 		}
 
+	case "websocket":
+		mux := http.NewServeMux()
+
+		connectionCount := 0
+
+		mux.HandleFunc("/", func(responseWriter http.ResponseWriter, request *http.Request) {
```

Comment:
in go the idiomatic name for the writer are `w`. For request it is usually `r` or `req`, but `request` isn't that uncommon.

Reply:
:+1: Switched to `w`